docs/plugin/errors.md

```diff
@@ -0,0 +1,506 @@
+class TestPlugin : public PluginHost::IPlugin, public PluginHost::JSONRPC {
+private:
+    class Notification : public RPC::IRemoteConnection::INotification {
+    public:
+        Notification() = delete;
+        explicit Notification(TestPlugin* parent)
+            : _parent(*parent)
+        {
+            ASSERT(parent != nullptr);
+        }
+
+        ~Notification() = default;
+        Notification(const Notification&) = delete;
+
+    public:
+        void Activated(RPC::IRemoteConnection* /* connection */) override
+        {
+        }
+        void Deactivated(RPC::IRemoteConnection* connectionId) override
+        {
+            _parent.Deactivated(connectionId);
+        }
+
+        BEGIN_INTERFACE_MAP(Notification)
+        INTERFACE_ENTRY(RPC::IRemoteConnection::INotification)
+        END_INTERFACE_MAP
+
+    private:
+        TestPlugin& _parent;
+    };
+
+public:
+    TestPlugin();
+    ~TestPlugin() override;
+
+    // Do not allow copy/move constructors
+    TestPlugin(const TestPlugin&) = delete;
+    TestPlugin& operator=(const TestPlugin&) = delete;
+
+    BEGIN_INTERFACE_MAP(TestPlugin)
+    INTERFACE_ENTRY(PluginHost::IPlugin)
+    INTERFACE_ENTRY(PluginHost::IDispatcher)
+    INTERFACE_AGGREGATE(Exchange::ITestPlugin, _testPlugin)
+    END_INTERFACE_MAP
+
+public:
+    // IPlugin methods
+    // ======================
+    const string Initialize(PluginHost::IShell* service) override;
+    void Deinitialize(PluginHost::IShell* service) override;
+    string Information() const override;
+
+private:
+    void Deactivated(RPC::IRemoteConnection* connection);
+
+private:
+    uint32_t _connectionId;
+    PluginHost::IShell* _service;
+    Exchange::ITestPlugin* _testPlugin;
+    Core::Sink<Notification> _notification;
+};
+
+```
+
+
+```c++ linenums="1" title="TestPlugin.cpp" hl_lines="5 24 46 65-76"
+TestPlugin::TestPlugin()
+    : _connectionId(0)
+    , _service(nullptr)
+    , _testPlugin(nullptr)
+    , _notification(this)
+{
+}
+
+TestPlugin::~TestPlugin()
+{
+}
+
+const string TestPlugin::Initialize(PluginHost::IShell* service)
+{
+    ASSERT(_service == nullptr);
+    ASSERT(_connectionId == 0);
+
+    string result = {};
+
+    _service = service;
+    _service->AddRef();
+
+    // Register for COM-RPC connection/disconnection notifications
+    _service->Register(&_notification);
+
+    // Instantiate the ITestPlugin interface (which will spawn the OOP side if running in OOP mode)
+    // Store connection ID in _connectionId
+    _testPlugin = _service->Root<Exchange::ITestPlugin>(_connectionId,
+                                                        2000,
+                                                        _T("TestPluginImplementation"));
+
+    if (!_testPlugin) {
+        // Error occurred, return non-empty string
+        result = "Failed to create ITestPlugin";
+    }
+
+    return result;
+}
+
+void TestPlugin::Deinitialize(PluginHost::IShell* service)
```

Comment:
I guess you discussed it with Pierre and we only want to discuss the connection notification but for a complete Deinitialize example some code would be missing. Let me take that up with Pierre.

Reply:
The main focus of the example is the connection notification but if the deinitialize() method isn't complete lets make is as accurate as possible. What's missing?

Reply:
I added it above in the mean time but I did that by editing my comment which was not that smart I guess :)

Indeed we realize the example is on the connection notification but I talked to Pierre and we also think to better have the  example complete in case people start to copy from it :)

See here for an example of the connection termination for OoP plugins that should be added: 
https://github.com/WebPlatformForEmbedded/ThunderNanoServicesRDK/blob/b1c0c7a47b8e758058b820c528dadedcf19965d6/Packager/Packager.cpp#L85-L102

Reply:
Great, thanks! Will update the example :)